REST-API-ADVANCED/advanced/src/main/java/com/epam/esm/tag/controller/TagController.java

```diff
@@ -0,0 +1,115 @@
+package com.epam.esm.tag.controller;
+
+import com.epam.esm.errorhandle.validation.Validator;
+import com.epam.esm.tag.entity.Tag;
+import com.epam.esm.tag.exception.TagInvalidRequestException;
+import com.epam.esm.tag.hateoas.TagHateoasSupport;
+import com.epam.esm.tag.service.TagService;
+import lombok.extern.slf4j.Slf4j;
+import org.springdoc.api.annotations.ParameterObject;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Pageable;
+import org.springframework.hateoas.CollectionModel;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@Slf4j
+@RestController
+@RequestMapping("/v1/api/tag")
+public class TagController {
+
+    private final TagService tagService;
+    private final TagHateoasSupport hateoasSupport;
+    private final Validator validator;
+
+    @Autowired
+    public TagController(TagService tagService, TagHateoasSupport hateoasSupport, Validator validator) {
+        this.tagService = tagService;
+        this.hateoasSupport = hateoasSupport;
+        this.validator = validator;
+    }
+
+    @GetMapping
+    public ResponseEntity<CollectionModel<Tag>> getAllTag(@ParameterObject Pageable paginationCriteria) {
+
+        log.debug("Request accepted getAllTag. Send request to service.");
+        List<Tag> allTag = tagService.getAllTag(paginationCriteria);
+
+        log.debug("Add hateoas to tags");
+        CollectionModel<Tag> tagsAndHateoas = hateoasSupport.addHateoasSupportToAllTag(allTag, paginationCriteria);
+
+        log.debug("Send response to client");
+        return ResponseEntity.ok(tagsAndHateoas);
+    }
+
+    @GetMapping("/{id}")
+    public ResponseEntity<Tag> getTagById(@PathVariable String id) {
+
+        log.debug("Request accepted getTagById. Validate id field.");
+        if (validator.isPositiveAndParsableId(id)) {
+
+            log.debug("Send request to service");
+            Tag serviceTag = tagService.getTag(Long.parseLong(id));
+
+            log.debug("Add hateoas to tag");
+            Tag tagAndHateoas = hateoasSupport.addHateoasSupportToSingleTag(serviceTag);
+
+            log.debug("Send response to client");
```

Comment:
Would be nice to have additional info in logs, like received tag, requested tag id etc. Without dynamic information logs are not really informative.

10 logs in console like:
```
Send response to client
Send response to client
Send response to client
Send response to client
...
```

Think what will help you identify potential problems and help to fix them faster.

Looks worse than:
```
Send response tag: Tag(id=1, name="someName") to client
Send response tag: Tag(id=2, name="someName2") to client
Send response tag: null to client
Send response tag: Tag(id=3, name="") to client
...
```

Reply:
done